libs/pmp-web/repository/repositories-statistics/feature/src/lib/containers/repositories-statistics/repositories-statistics.component.ts

```diff
@@ -0,0 +1,47 @@
+import {
+  ChangeDetectionStrategy,
+  ChangeDetectorRef,
+  Component,
+  OnDestroy,
+  OnInit
+} from '@angular/core';
+import { RepositoryStatisticsFacade } from '@pimp-my-pr/pmp-web/repository/data-access-repository-statistics';
+import { RepositoryStatistics } from '@pimp-my-pr/shared/domain';
+import { TableConfig } from '@pimp-my-pr/pmp-web/shared/domain';
+import { untilDestroyed } from 'ngx-take-until-destroy';
+import { RepositoriesStatisticsViewService } from './repositories-statistics-view.service';
+
+@Component({
+  selector: 'pimp-my-pr-repositories-statistics',
+  templateUrl: './repositories-statistics.component.html',
+  styleUrls: ['./repositories-statistics.component.scss'],
+  changeDetection: ChangeDetectionStrategy.OnPush,
+  providers: [RepositoriesStatisticsViewService]
+})
+export class RepositoriesStatisticsComponent implements OnInit, OnDestroy {
+  tableConfig: TableConfig<RepositoryStatistics[]>;
+
+  constructor(
+    private cdr: ChangeDetectorRef,
+    private repositoryStatisticsFacade: RepositoryStatisticsFacade,
+    private repositoriesStatisticsViewService: RepositoriesStatisticsViewService
+  ) {}
+
+  ngOnDestroy(): void {}
+
+  ngOnInit(): void {
+    this.repositoryStatisticsFacade.repositoryStatisticsCollection$
```

Comment:
I'm wondering if that should be placed inside a service.
Whar do you think?
@wjanaszek @krzysztofskorupkadev @matelener @patryk-zielinski93

Reply:
I think it should be something like this `this.presenter.setDataSource(this.repositoryStatisticsCollection$)`

Reply:
I also was not sure, but yesterday we agreed that with the facade, the component is communicating by itself, btw facade is also some kind of service.

Reply:
In my opinion, component could not have a dependency on a facade directly, just by the service, and that subscribe could be also placed in a presenter

Reply:
So we should put another part of abstraction, above all of the selectors which facade contain, inside of the presenter service?

Reply:
Nvm, I think this is a wrong lead

Reply:
I think it's better to have it separated, the component itself might take care of its input/output communication 

otherwise, we will often end up with both facade in the component and in component services which is an unnecessary dependency duplication

Reply:
I'm just wondering if it could be possible to make tableConfig also an observable.

Reply:
It's for sure possible and maybe even a better solution.

Reply:
separated -> facade only in the service, because very often that service will require data from the facade to perform business logic

Reply:
If Component would have every time only an observable which is to use in html we would never need to execute markForCheck manually, but anyone knows how to unsubscribe from observable inside a service?

Reply:
I think moving facade dependency to the service and getting rid of the facade in component could make sense only if we agree that service is a public field in the component and so are service's fields and we access them directly if needed. E.g. in the MyPresenter service we have "(public) itemsLoading$ = this.facade.itemsLoading$", then in component's ts file "(public) myPresenter: MyPresenter" and then somewhere in HTML we can access it like '<div *ngIf=myPresenter.itemsLoading$ | async" ... '.
Another thing is that we would still need to wrap every facade method in presenter method though, which I find cumbersome.

Reply:
The only solution I see without research is to execute some method in the service from onDestroy in component.

Reply:
Ok with @matelener we confirmed that ngOnDestroy works for services provided in component

Reply:
I am inclined to not make the Presenter a proxy for facade to not make a lasagna architecture. I'm just thinking if it is good to move that subscription to the Presenter and make the formGroup an observable